README.md

```diff
@@ -0,0 +1,178 @@
+# Go Middleware Generator
+
+This tool can generate a generic middleware, that injects hooks around
+the invocations of given interface' methods.
+
+# Description
+
+The generated middleware wraps around another implementation of the same interface
+and calls specific hooks at specific points in the execution.
+
+Given the following interface and `go:generate` command
+```go
+//go:generate go run github.com/Makpoc/gomigen/cmd/gomigen . Interface
+type Interface interface {
+    ProcessOne(string) (string, error)
+    ProcessTwo(context.Context, int) bool
+}
+```
+
+running `go:generate` will produce the following Middleware:
+
+* Struct:
```

Comment:
NIT: The other bullet points don't have a `:` at the end

Reply:
Also, if you want to use bullet points, then you need to tab everything that is meant to reside inside. Otherwise consider a different paragraph approach.

Reply:
Changed to headers. I used ### because # and ## look too similar. Not sure if it's just my font, but I like the visual separation between # and ### better